monix-eval/shared/src/main/scala/monix/eval/instances/CatsSemigroupKForTask.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.eval.instances
+import cats.SemigroupK
+import monix.eval.Task
+
+class CatsSemigroupKForTask extends SemigroupK[Task] {
+
+  def combineK[A](a: Task[A], b: Task[A]): Task[A] =
+    Task.catsAsync.handleErrorWith(a)(_ => b)
```

Comment:
I think we could use `a.onErrorHandleWith(_ => b)` directly but I'm not sure if it makes difference though